eth/eventservices/rewardservice_test.go

```diff
@@ -0,0 +1,156 @@
+		cancelWorker: cancel,
+	}
+	errC := make(chan error)
+	go func() { errC <- rs.Start(ctx) }()
+	time.Sleep(1 * time.Second)
+	assert.True(rs.working)
+	cancel()
+	err := <-errC
+	assert.Nil(err)
+}
+
+func TestStop(t *testing.T) {
+	assert := assert.New(t)
+	rs := RewardService{
+		working: false,
+	}
+	assert.EqualError(rs.Stop(), ErrRewardServiceStopped.Error())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	rs = RewardService{
+		tw:           &stubTimeWatcher{},
+		cancelWorker: cancel,
+	}
+	go rs.Start(ctx)
+	time.Sleep(1 * time.Second)
+	require.True(t, rs.working)
+	rs.Stop()
+	assert.False(rs.working)
+}
+
+func TestIsWorking(t *testing.T) {
+	assert := assert.New(t)
+	rs := RewardService{
+		working: false,
+	}
+	assert.False(rs.IsWorking())
+	rs.working = true
+	assert.True(rs.IsWorking())
+}
+
+func TestReceiveRoundEvent_TryReward(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+	eth := &eth.MockClient{}
+	tw := &stubTimeWatcher{}
+	ctx := context.Background()
+	rs := RewardService{
+		client: eth,
+		tw:     tw,
+	}
+
+	go rs.Start(ctx)
+	time.Sleep(1 * time.Second)
+	require.True(rs.IsWorking())
+
+	// Happy case , check that reward was called
+	// Assert that no error was logged
+	eth.On("CurrentRound").Return(big.NewInt(10), nil)
+	eth.On("CurrentRoundInitialized").Return(true, nil)
+	eth.On("Account").Return(accounts.Account{})
+	eth.On("GetTranscoder").Return(&lpTypes.Transcoder{
+		LastRewardRound: big.NewInt(1),
+	}, nil)
+	eth.On("IsActiveTranscoder").Return(true, nil)
+	eth.On("Reward").Return(&types.Transaction{}, nil).Times(1)
+	eth.On("CheckTx").Return(nil).Times(1)
+	eth.On("GetTranscoderEarningsPoolForRound").Return(&lpTypes.TokenPools{}, nil)
+
+	errorLogsBefore := glog.Stats.Error.Lines()
+	infoLogsBefore := glog.Stats.Info.Lines()
+
+	tw.roundSink <- types.Log{}
+	time.Sleep(1 * time.Second)
+
+	eth.AssertNumberOfCalls(t, "Reward", 1)
+	eth.AssertNumberOfCalls(t, "CheckTx", 1)
+	eth.AssertNotCalled(t, "ReplaceTransaction")
+
+	errorLogsAfter := glog.Stats.Error.Lines()
+	infoLogsAfter := glog.Stats.Info.Lines()
+	assert.Equal(int64(0), errorLogsAfter-errorLogsBefore)
+	assert.Equal(int64(1), infoLogsAfter-infoLogsBefore)
+
+	// Test for transaction time out error
+	// Call replace transaction
+	eth.On("Reward").Return(&types.Transaction{}, nil).Times(1)
+	eth.On("CheckTx").Return(context.DeadlineExceeded).Times(1)
+	eth.On("ReplaceTransaction").Return(&types.Transaction{}, nil)
+
+	errorLogsBefore = glog.Stats.Error.Lines()
+	infoLogsBefore = glog.Stats.Info.Lines()
+
+	tw.roundSink <- types.Log{}
+	time.Sleep(1 * time.Second)
+
+	eth.AssertNumberOfCalls(t, "Reward", 2)
+	eth.AssertNumberOfCalls(t, "CheckTx", 2)
+	eth.AssertNumberOfCalls(t, "ReplaceTransaction", 1)
+
+	errorLogsAfter = glog.Stats.Error.Lines()
+	infoLogsAfter = glog.Stats.Info.Lines()
+	assert.Equal(int64(0), errorLogsAfter-errorLogsBefore)
+	assert.Equal(int64(1), infoLogsAfter-infoLogsBefore)
```

Comment:
Should log 2 info statements here 
1. "replacing transaction" 
2. "called reward" 

But only 1 is picked up by `glog`

Reply:
See my other comment about the current behavior when `ReplaceTransaction()` is called - the log statement for "Called reward..." will never be reached because `tryReward()` will return before then.